### PR TITLE
Update pedantic

### PR DESCRIPTION
**Pull request by listepo**

### CODE REVIEW

File: .travis.yml
Comment (listepo) [inline .travis.yml:25]: Required for spread-collections, but I can roll back if it is critical

**Comment by listepo** [comment]
@sigurdm do you have any comments?

File: protoc_plugin/analysis_options.yaml
Comment (kevmoo) [inline protoc_plugin/analysis_options.yaml:14]: keep a newline at the end of the file!

Comment (listepo) [reply protoc_plugin/analysis_options.yaml]: done

**Comment by mit-mit** [comment]
Let's not merge this before @nichite has had a chance to review how these lints might impact rolling this into google3.

File: protoc_plugin/lib/indenting_writer.dart
Comment (nichite) [inline protoc_plugin/lib/indenting_writer.dart:9]: Would it be better to just inline this instead of creating the private typedef, especially since the signature is so simple? I'm thinking of the avoid_private_typedef_functions lint and this preference: https://critique-ng.corp.google.com/cl/326560963

File: protoc_plugin/CHANGELOG.md
Comment (nichite) [inline protoc_plugin/CHANGELOG.md:4]: Are there more changes planned to go into some larger release for this package? It seems like the SDK jump to 2.3.0 is a pretty large change to put in the dev release, was wondering if it's worth a version bump

File: protoc_plugin/lib/bazel.dart
Comment (nichite) [inline protoc_plugin/lib/bazel.dart:46]: Can you use the `OnError` typedef defined in `options.dart`?

**Comment by nichite** [comment]
Overall the changes LGTM. I don't think any of the small tweaks made here to make the analyzer happy will cause issues rolling into Google3. I do want to make sure @iinozemtsev is aware of these changes since he's been working on syncing the internal and external repos.

File: protoc_plugin/CHANGELOG.md
Comment (listepo) [reply protoc_plugin/CHANGELOG.md:4]: I do not plan, but I have seen such changes in others, like https://github.com/dart-lang/protobuf/pull/394

File: protoc_plugin/lib/indenting_writer.dart
Comment (listepo) [reply protoc_plugin/lib/indenting_writer.dart:9]: You're right

File: protoc_plugin/lib/bazel.dart
Comment (listepo) [reply protoc_plugin/lib/bazel.dart:46]: You're right

**Review by iinozemtsev — APPROVED** [review]
LG, this won't complicate syncing with internal version

**Comment by kevmoo** [comment]
Who wants to hit the merge button?

**Pull request merged**
